assets/js/modules/analytics-4/components/common/PropertySelect.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * GA4 Property Select component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { Select, Option } from '../../../../material-components';
+import ProgressBar from '../../../../components/ProgressBar';
+import { STORE_NAME } from '../../datastore/constants';
+import { MODULES_ANALYTICS, PROPERTY_CREATE } from '../../../analytics/datastore/constants';
+import { isValidAccountID } from '../../../analytics/util';
+import { trackEvent } from '../../../../util';
+const { useSelect, useDispatch } = Data;
+
+export default function PropertySelect() {
+	const {
+		accountID,
+		properties,
+		isResolvingProperties,
+	} = useSelect( ( select ) => {
+		const data = {
+			accountID: select( MODULES_ANALYTICS ).getAccountID(),
+			properties: [],
+			isResolvingProperties: false,
+		};
+
+		if ( data.accountID ) {
+			data.properties = select( STORE_NAME ).getProperties( data.accountID );
+			data.isResolvingProperties = select( STORE_NAME ).isResolving( 'getProperties', [ data.accountID ] );
+		}
+
+		return data;
+	} );
+
+	const propertyID = useSelect( ( select ) => select( STORE_NAME ).getPropertyID() );
+	const hasResolvedAccounts = useSelect( ( select ) => select( MODULES_ANALYTICS ).hasFinishedResolution( 'getAccounts' ) );
+
+	const { selectProperty } = useDispatch( STORE_NAME );
+	const onChange = useCallback( ( index, item ) => {
+		const newPropertyID = item.dataset.value;
+		if ( propertyID !== newPropertyID ) {
+			selectProperty( newPropertyID, item.dataset.internalId ); // eslint-disable-line sitekit/acronym-case
+			trackEvent( 'analytics_setup', 'property_change', newPropertyID );
```

Comment:
Not sure that we need it for ga4. I believe we will have a separate ticket for ga4 tracking.

Reply:
Good spot, but I double checked the ACs and I think we are good to keep this.